immuni_exposure_ingestion/models/validators.py

```diff
@@ -7,58 +9,152 @@
 from immuni_common.models.mongoengine.temporary_exposure_key import TemporaryExposureKey
 from immuni_exposure_ingestion.core import config
 
+_LOGGER = logging.getLogger(__name__)
+
+_ROLLING_PERIOD_MIN = 0
+_ROLLING_PERIOD_MAX = 144
+
 
 class TekListValidator(Validator):
     """
     A validator for a list of TEKs.
     """
 
-    def __call__(self, value: List[TemporaryExposureKey]) -> None:
+    def __call__(self, teks: List[TemporaryExposureKey]) -> None:
         """
-        Validations to be performed according to Apple's documentation:
+        Perform the validation of a given list of TEKs, both on punctual value and as a whole.
 
-        - Any ENIntervalNumber values from the same user are not unique.
-        - The period of time covered by the data file exceeds 14 days.
-        - Any keys in the file have overlapping time windows.
-        - The TEKRollingPeriod value is not 144.
+        NOTE: The validation of a single TEKs could be performed at field level.
+          We expect (some of) these validations to change in the future. Having them all together
+          here aims at easing any future change in the validation steps.
 
-        :param value: the value of the schema being validated.
+        :param teks: the list of TEKs to be validated.
         """
 
-        if len(value) == 0:
-            return
-
-        # NOTE: This kind of validation (rolling_period == 144) could be performed at field level.
-        #   We expect all of these validations to change in the future, so having them all together
-        #   here is done on purpose to make things simpler in case of changes in validation.
-        #   We would like to keep field validation more flexible as not to spread strict validation
-        #   logic in many places.
+        n_teks = len(teks)
+        _LOGGER.info("Validating TEKs.", extra=dict(n_teks=n_teks))
 
-        if any(tek.rolling_period != 144 for tek in value):
-            raise ValidationError("Some rolling values are not 144.")
+        if n_teks == 0:
+            return
 
-        if (n_keys := len(value)) > config.MAX_KEYS_PER_UPLOAD:
+        if n_teks > config.MAX_KEYS_PER_UPLOAD:
             raise ValidationError(
-                f"Too many TEKs. (actual: {n_keys}, max_allowed: {config.MAX_KEYS_PER_UPLOAD})."
+                f"Too many TEKs. (actual: {n_teks}, max_allowed: {config.MAX_KEYS_PER_UPLOAD})."
             )
 
-        sorted_teks = sorted(value, key=operator.attrgetter("rolling_start_number"))
+        valid_teks = self._filter_valid_teks(teks)
+
+        sorted_teks = sorted(
+            valid_teks, key=operator.attrgetter("rolling_start_number")
+        )
         rolling_start_numbers = [t.rolling_start_number for t in sorted_teks]
 
+        # TEKs with the same rolling_start_number can exist when considering a rolling_period < 144.
```

Comment:
In my understanding - but please countercheck on the A/G specifications - when teks with `rolling period < 144` will be released, you won't have multiple TEKs with the same `rolling_start_number`. 
The `rolling_start_number` is `epoch_timestamp / 600`, and it is indicating when the tek started being broadcasted, whilst the `rolling_period` is indicating for how much time the tek was broadcasted. At the moment, with all the `rolling_period` set to 144, the rolling start number is always set to the `epoch_timestamp_at_midnight/600`. When the SDK will rotate the key during the same day, generating rolling periods < 144, it will generate two teks, having two different rolling start number, with the latter tek indicating when it started being broadcasted, so that `rolling_start_number1 + rolling_period1` should be equal to `rolling_start_number2` 😊 Btw, please countercheck this to be 100% sure 😊

Reply:
I read
```
As a result, if this method is called multiple times, several keys with different rollingPeriod and identical rollingStartNumber can be released for those days with multiple calls
```

Reply:
Thus
`rolling_start_number1 + rolling_period1 should be equal to rolling_start_number2`
may not always hold true.